brax/training/agents/ppo/train.py

```diff
@@ -231,12 +231,17 @@ def train(
   key_envs = jnp.reshape(key_envs,
                          (local_devices_to_use, -1) + key_envs.shape[1:])
   env_state = reset_fn(key_envs)
+  ndarray_obs = isinstance(env_state.obs, jnp.ndarray) # Check whether observations are in dictionary form.
+  if not ndarray_obs and normalize_observations:
+    assert "state" in env.observation_size, "Observation normalisation only supported for states."
+
+  obs_shape = env_state.obs.shape[-1] if ndarray_obs else env.observation_size
```

Comment:
Let's not rely on env.observation_size, I believe in many cases, that calls an env.step

Reply:
switched to using jax.tree_util.tree_map(lambda x: x.shape, env_state.obs)